R/bm-write-csv.R

```diff
@@ -0,0 +1,74 @@
+#' Benchmark CSV writing
+#'
+#' @section Parameters:
+#' * `source` A CSV file path to write to
+#' * `writer` One of `c("arrow", "data.table", "vroom", "readr",)`
+#' * `input` One of `c("arrow_table", "data_frame")`
+#'
+#' @export
+write_csv <- Benchmark(
+  "write_csv",
+  setup = function(source = names(known_sources),
+                   writer = c("arrow", "data.table", "vroom", "readr", "utils"),
+                   input = c("arrow_table", "data_frame")) {
+    writer <- match.arg(writer)
+    input <- match.arg(input)
+
+    # source defaults are retrieved from the function definition (all available
+    # known_sources) and then read the source in as a data.frame
+    source <- ensure_source(source)
+    df <- read_source(source, as_data_frame = match.arg(input) == "data_frame")
+
+    # Map string param name to functions
+  BenchEnvironment(
+      write_csv_func = get_csv_writer(writer),
+      source = source,
+      df = df
+    )
+  },
+  # delete the results before each iteration
+  before_each = {
+    result_file <- tempfile(fileext = ".csv")
+  },
+  # the benchmark to run
+  run = {
+    write_csv_func(df, result_file)
+  },
+  # after each iteration, check the dimensions and delete the results
+  after_each = {
+    stopifnot(file.exists(result_file))
+    unlink(result_file)
+  },
+  valid_params = function(params) {
+    ## Only arrow fns will accept an arrow_table
+    drop <- ( params$input == "arrow_table" & params$writer != "arrow" )
+    params[!drop,]
+  },
+  packages_used = function(params) {
+    params$writer
```

Comment:
We might want to strip out `"utils"` here. I suspect it'll work fine, but we'll get the warning `package ‘utils’ is a base package, and should not be updated`

Reply:
Separately from that, I wonder if we should call this `"base"`. `"utils"` is most accurate, but I suspect most folks would call it `"base"`. I'm not the fence, what do you think?

Reply:
Yeah actually for intent I think `"base"` is probably best. I was sort of on auto-pilot which led me to `"utils"`.